### PR TITLE
refactor: create util folder, move nap_sym to napi/sym, move http_cache to cache folder

### PR DESCRIPTION
**Pull request by dsherret**
Probably last cleanup refactor.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -5,7 +5,7 @@ resolver = "2"
 members = [
   "bench_util",
   "cli",
-  "cli/napi_sym",
+  "cli/napi/sym",
```
Comment (dsherret) [inline Cargo.toml:8]: Buries napi_sym in the napi folder.

File: cli/args/mod.rs [hunk 2]
```diff
@@ -40,16 +44,130 @@ use deno_runtime::inspector_server::InspectorServer;
...
 use crate::cache::DenoDir;
-use crate::file_fetcher::get_root_cert_store;
-use crate::file_fetcher::CacheSetting;
-use crate::fs_util;
+use crate::util::fs::canonicalize_path_maybe_not_exists;
 use crate::version;
 
+/// Indicates how cached source files should be handled.
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum CacheSetting {
```
Comment (dsherret) [inline cli/args/mod.rs:58]: Moved from file_fetcher and removed http_cache logic from this enum.

File: cli/args/mod.rs [hunk 3]
```diff
@@ -40,16 +44,130 @@ use deno_runtime::inspector_server::InspectorServer;
...
+  /// default behavior of the CLI.
+  Use,
+}
+
+impl CacheSetting {
+  pub fn should_use_for_npm_package(&self, package_name: &str) -> bool {
+    match self {
+      CacheSetting::ReloadAll => false,
+      CacheSetting::ReloadSome(list) => {
+        if list.iter().any(|i| i == "npm:") {
```
Comment (dsherret) [inline cli/args/mod.rs:83]: I did do a change on this line. It was previously doing `list.contains(&"npm:".to_string())`, so this reduces an allocation.

File: cli/args/mod.rs [hunk 4]
```diff
@@ -40,16 +44,130 @@ use deno_runtime::inspector_server::InspectorServer;
...
+        true
+      }
+      _ => true,
+    }
+  }
+}
+
+/// Create and populate a root cert store based on the passed options and
+/// environment.
+pub fn get_root_cert_store(
```
Comment (dsherret) [inline cli/args/mod.rs:99]: Moved from file_fetcher.rs. This is used below in CliOptions.

File: cli/file_fetcher.rs [hunk 5]
```diff
@@ -654,6 +494,41 @@ impl FileFetcher {
     .boxed()
   }
 
+  /// Returns if the cache should be used for a given specifier.
+  fn should_use_cache(&self, specifier: &ModuleSpecifier) -> bool {
```
Comment (dsherret) [inline cli/file_fetcher.rs:498]: Moved from `CacheSetting`. This can be private here.

File: cli/tools/bench.rs [hunk 6]
```diff
@@ -469,6 +470,19 @@ async fn bench_specifiers(
   Ok(())
 }
 
+/// Checks if the path has a basename and extension Deno supports for benches.
+fn is_supported_bench_path(path: &Path) -> bool {
```
Comment (dsherret) [inline cli/tools/bench.rs:474]: This was specific to benchmarking, so moved this here.

File: cli/tools/test.rs [hunk 7]
```diff
@@ -1184,6 +1185,44 @@ async fn test_specifiers(
   Ok(())
 }
 
+/// Checks if the path has a basename and extension Deno supports for tests.
+fn is_supported_test_path(path: &Path) -> bool {
```
Comment (dsherret) [inline cli/tools/test.rs:1189]: Similarly, these were for testing, so moved these functions here.

File: cli/util/fs.rs [hunk 8]
```diff
@@ -163,77 +163,6 @@ pub fn resolve_from_cwd(path: &Path) -> Result<PathBuf, AnyError> {
   Ok(normalize_path(&resolved_path))
 }
 
-/// Checks if the path has extension Deno supports.
-pub fn is_supported_ext(path: &Path) -> bool {
```
Comment (dsherret) [inline cli/util/fs.rs:167]: Split the previous fs_util.rs file into `util/fs.rs` and `util/path.rs`

File: cli/util/path.rs [hunk 9]
```diff
@@ -0,0 +1,452 @@
+use std::borrow::Cow;
+use std::path::Path;
+use std::path::PathBuf;
+
+use deno_ast::ModuleSpecifier;
+use deno_core::error::uri_error;
+use deno_core::error::AnyError;
+
+/// Checks if the path has extension Deno supports.
+pub fn is_supported_ext(path: &Path) -> bool {
```
Comment (dsherret) [inline cli/util/path.rs:10]: No changes to the logic in this file. It was just a straight extract out from fs_util.rs.

File: Cargo.toml (hunk 1 above)
Comment (dsherret) [reply Cargo.toml:8]: Yeah, should be ok. I can do the release this week 😅

Comment (bartlomieju) [reply Cargo.toml:8]: Sounds good :)

**Review by bartlomieju — APPROVED** [review]
LGTM, nice cleanup

**Pull request merged**